spec/queries/intent_metrics_query_spec.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+require 'rails_helper'
+RSpec.describe IntentMetricsQuery do
+  let(:draw) { FactoryGirl.create(:draw) }
+
+  it 'returns a hash with metric values for all intent strings' do
+    create_intent_data(draw)
+    expected = { 'off_campus' => 1, 'on_campus' => 2, 'undeclared' => 3 }
+
+    result = described_class.call(draw)
+
+    expect(result).to eq(expected)
+  end
+
+  it 'defaults to zero for intents with no students' do
+    FactoryGirl.create(:user, draw: draw, intent: 'off_campus')
+    expected = { 'off_campus' => 1, 'on_campus' => 0, 'undeclared' => 0 }
+
+    result = described_class.call(draw)
+
+    expect(result).to eq(expected)
+  end
+
+  it 'ignores students from other draws' do
+    FactoryGirl.create(:user, intent: 'off_campus')
+    expected = { 'off_campus' => 0, 'on_campus' => 0, 'undeclared' => 0 }
+
+    result = described_class.call(draw)
+
+    expect(result).to eq(expected)
+  end
+
+  def create_intent_data(draw)
```

Comment:
Yea, I did that for the other spec but that was a later refactor. I'll update this one now.